x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/model/rule_assets/prebuilt_rule_asset.ts

```diff
@@ -6,13 +6,70 @@
  */
 
 import * as z from 'zod';
+import type {
+  EqlRuleCreateFields,
+  QueryRuleCreateFields,
+  SavedQueryRuleCreateFields,
+  ThresholdRuleCreateFields,
+  ThreatMatchRuleCreateFields,
+  MachineLearningRuleCreateFields,
+  NewTermsRuleCreateFields,
+  EsqlRuleCreateFields,
+} from '../../../../../../common/api/detection_engine/model/rule_schema';
 import {
   RuleSignatureId,
   RuleVersion,
   BaseCreateProps,
   TypeSpecificCreateProps,
 } from '../../../../../../common/api/detection_engine/model/rule_schema';
 
+type BaseFields = keyof BaseCreateProps;
+type SpecificFields =
+  | keyof Omit<EqlRuleCreateFields, 'type'>
+  | keyof Omit<QueryRuleCreateFields, 'type'>
+  | keyof Omit<SavedQueryRuleCreateFields, 'type'>
+  | keyof Omit<ThresholdRuleCreateFields, 'type'>
+  | keyof Omit<ThreatMatchRuleCreateFields, 'type'>
+  | keyof Omit<MachineLearningRuleCreateFields, 'type'>
+  | keyof Omit<NewTermsRuleCreateFields, 'type'>
+  | keyof Omit<EsqlRuleCreateFields, 'type'>;
+
+type AllRuleProperties = BaseFields | SpecificFields;
+
+/**
+ * The PrebuiltRuleAsset is created out of the Rule schema types defined in our OpenAPI schemas.
+ * However, we don't need all the fields in the rule schemas to be present in the PrebuiltRuleAsset.
+ * We manually exclude those unwanted fields here.
+ *
+ * Reference ticket:
+ * https://github.com/elastic/kibana/issues/180393
+ *
+ */
+type OmittedProperties = Extract<
+  AllRuleProperties,
+  | 'actions'
+  | 'throttle'
+  | 'meta'
+  | 'output_index'
+  | 'namespace'
+  | 'alias_purpose'
+  | 'alias_target_id'
+  | 'outcome'
+>;
+
+const PROPERTIES_TO_OMIT: Record<OmittedProperties, true> = {
+  actions: true,
+  throttle: true,
+  meta: true,
+  output_index: true,
+  namespace: true,
+  alias_purpose: true,
+  alias_target_id: true,
+  outcome: true,
+};
```

Comment:
What about the `response_actions` field? It's not a part of `BaseCreateProps` because it's a type-specific field. `QueryRuleOptionalFields` and `SavedQueryRuleOptionalFields` contain it. This means that it is included to `PrebuiltRuleAsset` together with other `TypeSpecificCreateProps`.

Reply:
Thanks for catching this, I had indeed missed `response_actions`. Take a look at `x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/model/rule_assets/prebuilt_rule_asset.ts` now, I had to do some more type wrangling to omit a type that is part of the `TypeSpecificCreateProps`. Similar to what I had done initially, but now using `zodMaskFor`